### Problema

Se desea diseñar un intercambiador de calor de tubos y coraza, con queroseno y petróleo como fluidos de trabajo. La caída de presión en el lado coraza debe ser de 15kPa y 40 kPa para el lado tubos. 

Considerar que el material del intercambiador es acero al carbón (k = 43W/mK). 

#### Consideraciones
<ul>

<li>El sistema se encuentra bien aislado.</li>
<li>El flujo es contracorriente.</li>
</ul>

#### Propiedades

<table title = "Propiedades">
<tr>
    <th scope = "col"> </th>
    <th scope = "col"> Queronseno </th>
    <th scope = "col"> Petróleo </th>
</tr>
<tr>
    <td> Cp [J/kg°C]</td>
    <td> 2470 </td>
    <td> 2050 </td>
</tr>
<tr>
    <td> k [W/m°C] </td>
    <td> 0.132 </td>
    <td> 0.133 </td>
</tr>
<tr>
    <td> μ [mPa s]</td>
    <td> 0.4 </td>
    <td> 3.6 </td>
</tr>
<tr>
    <td> ρ [kg/m3] </td>
    <td> 730 </td>
    <td> 830 </td>
</tr>
<tr>
    <td> m [kg/s] </td>
    <td> 25 </td>
    <td> ¿? </td>
</tr>
<tr>
    <td> Ti [°C] </td>
    <td> 200 </td>
    <td> 35 </td>
</tr>
<tr>
    <td> To [°C] </td>
    <td> 95.16 </td>
    <td> 74.94 </td>
</tr>
<tr>
    <td> R [m2°C/W] </td>
    <td> 0.0002 </td>
    <td> 0.0005 </td>
</tr>
</table>



#### Anexo

<ul>
<li>Gráficos de jh y jf en función de Re para tubos y para coraza, tomado de [1]. </li>

<img src = "./Graficos/jh lado tubos.png" alt = "Gráfico jh lado tubos">
<p> Fig. 1. Valores de jh en función de Re y la relación L/D para lado tubos</p>
<img src = "./Graficos/jf lado tubos.png" alt = "Gráfico jf lado tubos">
<p> Fig. 2. Valores de jf para lado tubos </p>
<img src = "./Graficos/jh lado coraza.png" alt = "Gráfico jh lado coraza">
<p> Fig. 3. Valores de jh para lado coraza </p>
<img src = "./Graficos/jf lado coraza.png" alt = "Gráfico jf lado coraza">
<p> Fig. 4. Valores de jf para lado coraza </p>

<li> Valores típicos de U para intercambiador de tubos y coraza, tomado de [1].</li>
<img src = "./Graficos/Valores_tipicos_U_ITyC.png" alt = "Tabla de valores de U para varios sistemas, ITyC.">
<p> Fig. 5. Tabla de valores de U para varios sistemas, ITyC. </p>


In [1]:
import objetos as obj
import Diseño_tubos_coraza as dtc
import matplotlib.pyplot as plt

Se crean objetos con referencia a los fluidos y el material.

In [2]:
# creación de objetos

tuberia_acero = obj.Tuberia(
    material = "Acero al carbón",
    do = 0.019,
    espesor = 0.001,
    L = 8.61,
    k = 43 
)

FlujoCaliente = obj.FlujoFluido(
    fluido = "queroseno",
    temperatura_entrada= 200,
    temperatura_salida= 95.16,
)

FlujoFrio = obj.FlujoFluido(
    fluido = "petróleo",
    temperatura_entrada= 35,
    temperatura_salida= 74.9,
)

propiedadesQueroseno = obj.Propiedades(
    fluido = "queroseno",
    temp_entrada_salida = [200,95.16],
    temperaturas_propiedades = "NA",
    densidad = 730,
    viscosidad = 0.4e-3,
    cp = 2470,
    coeficiente_conductivo = 0.132,
    numero_prandtl = 0.4e-3*2470/0.132,
    Rf = 0.0002,
    flujo_masico = 25
)

propiedadesPetroleo = obj.Propiedades(
    fluido = "Petroleo",
    temp_entrada_salida = [35,74.94],
    temperaturas_propiedades = "NA",
    densidad = 830,
    viscosidad = 3.6e-3,
    cp = 2050,
    coeficiente_conductivo = 0.133,
    numero_prandtl = 3.6e-3*2050/0.133,
    Rf = 0.0005,
    flujo_masico = 0
)

intercambiador1 = obj.Intercambiador(

)


Se determinan los parámetros faltantes, como es Q y el flujo másico del petróleo por medio de un balance de energía

In [3]:
Q, m2 = obj.balance_energia(
    m = [25,0],
    Cp = [2470,2050],
    Ti = [200,35],
    To = [95.16,74.94],
    calc = "m"
)

print("Q = ",Q ,"\nm2 = ", m2)

propiedadesPetroleo.flujo_masico = m2

Se desconoce el flujo másico del fluido que se calienta.
Q =  6473870.0 
m2 =  79.0682365010931


Con base en la resistencia de ensuciamiento se decide que el queroseno, que es el que menos ensucia, es el fluido que va por coraza; mientras que el petróleo va por tubos.


In [4]:
FlujoTubos = propiedadesPetroleo
FlujoCoraza = propiedadesQueroseno

## Iteración 1

Se procede ahora a proponer un coeficiente global de transferencia de calor para el intercambiador, el cual, con base en la figura 5 y teniendo en cuenta que son aceites pesados (Heavy oils) los fluidos de trabajo, se escoge arbitrariamente un valor entre 50 y 300 W/m2°C.

Además, se establecen características del diseño, como el número de pasos por tubos y por coraza, el corte de bafle y el tipo de pitch.

In [5]:
U = 291

pasos_tubos = 1
pasos_coraza = 1
corte_bafle = '15%'

tipo_pitch = "triangular"

Se evalúan las porpiedades para el lado tubos

In [6]:
obj.lado_tubos(
    Q = Q,
    U = U, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador1
    )


Para el lado tubos:


(226.5690672072488, 541.6321489955797, 7084.4857058898515)

Para el lado coraza

In [7]:
obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador1.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador1
)

(2002.815051782778, 245205.3155494682)

In [8]:
print(intercambiador1)


LADO TUBOS: 


    Pasos: 1
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2
    Número de tubos: 541.6321489955797 tubos
    Longitud tubos: 8.61 m
    Diámetro exterior: 0.019 m
    Espesor: 0.001 m 
    Tubos por paso: 541.6321489955797 tubos
    velocidad en lado tubos: 0.7748752719680598 m/s
    Reynolds en tubos: 3037.0805798525903

    jh: 0.0025
    jf: 0.0064
    h lado tubos (hi): 226.5690672072488 W/m2K
    ΔP lado tubos: 7084.4857058898515 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 0.6118568737572134 m
    Diámetro interno coraza: 0.6732323606103877 m
    Separación de bafles: 0.2692929442441551 m
    Pitch de los tubos: 0.02375 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.03625934490984282 m2
    Diámetro hidráulico: 0.01349095 m
    Velocidad lado coraza: 0.9444896323311487 m/s
    Re: 23254.263889668688
    Corte de bafle: 15%

    jh: 0.0045
    jf: 0.059
    h lado coraza 

Con la información obtenida, se calcula el coeficiente global de transferencia de calor y se determina el error asociado.


In [9]:
U_calculado = obj.coefiente_global_U(
    h_conv = [intercambiador1.hi,intercambiador1.h_coraza],
    k = tuberia_acero.k,
    di = tuberia_acero.di,
    do = tuberia_acero.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U,
    valor_calculado = U_calculado
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador1.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador1.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 160.88515785005987 W/m2K
Para el valor calculado = 160.88515785005987, %error = 44.71300417523716%
Para el valor calculado = 7084.4857058898515, %error = 82.28878573527538%
Para el valor calculado = 245205.3155494682, %error = 1534.7021036631213%


Podemos observar que los valores obtenidos se salen del rango de aceptación (%e para U = 30%, %e ΔP = 10%). Se proponen 2 pasos por el lado tubos para aumentar la tasa de transferencia de calor y la caída de presión. Para el lado coraza, el corte de bafle se aumenta de 15 a 45% para disminuir la excesiva caída de presión.

## Iteración 2

In [10]:
pasos_tubos = 2
corte_bafle = "45%"

intercambiador2 = obj.Intercambiador()

In [11]:
obj.lado_tubos(
    Q = Q,
    U = U, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador2
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador2.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador2
)

print(intercambiador2)

Para el lado tubos:

LADO TUBOS: 


    Pasos: 2
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2
    Número de tubos: 541.6321489955797 tubos
    Longitud tubos: 8.61 m
    Diámetro exterior: 0.019 m
    Espesor: 0.001 m 
    Tubos por paso: 270.8160744977898 tubos
    velocidad en lado tubos: 1.5497505439361197 m/s
    Reynolds en tubos: 6074.161159705181

    jh: 0.0035
    jf: 0.0055
    h lado tubos (hi): 634.3933881802967 W/m2K
    ΔP lado tubos: 49406.65558778859 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 0.6180017537735323 m
    Diámetro interno coraza: 0.6795469622127572 m
    Separación de bafles: 0.2718187848851029 m
    Pitch de los tubos: 0.02375 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.03694272590820693 m2
    Diámetro hidráulico: 0.01349095 m
    Velocidad lado coraza: 0.9270180935635229 m/s
    Re: 22824.097417583478
    Corte de bafle: 45%

    jh: 0.003
    jf: 0.028
 

In [12]:
U_calculado2 = obj.coefiente_global_U(
    h_conv = [intercambiador2.hi,intercambiador2.h_coraza],
    k = tuberia_acero.k,
    di = tuberia_acero.di,
    do = tuberia_acero.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U,
    valor_calculado = U_calculado2
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador2.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador2.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 302.2778158976255 W/m2K
Para el valor calculado = 302.2778158976255, %error = 3.8755381091496517%
Para el valor calculado = 49406.65558778859, %error = 23.51663896947148%
Para el valor calculado = 112103.17987557504, %error = 647.3545325038335%


Los resultados obtenidos se acercan a lo requerido pues en términos de transferencia de calor el diseño es aceptable; sin embargo, no cumple para la caída de presión. Se propone un redimensionamiento de los tubos utilizados, disminuyendo su longitud y aumentando su diámetro, de forma que la caída de presión en ambos lados disminuya.

## Iteración 3

In [13]:
tuberia_acero2 = obj.Tuberia(
    material = "Acero al carbón",
    do = 0.020,
    espesor = 0.001,
    L = 6,
    k = 43
)

intercambiador3 = obj.Intercambiador()


In [14]:
obj.lado_tubos(
    Q = Q,
    U = U, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero2, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador3
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero2,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador3.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador3
)

print(intercambiador3)



Para el lado tubos:

LADO TUBOS: 


    Pasos: 2
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2
    Número de tubos: 738.3800271182239 tubos
    Longitud tubos: 6 m
    Diámetro exterior: 0.02 m
    Espesor: 0.001 m 
    Tubos por paso: 369.19001355911195 tubos
    velocidad en lado tubos: 1.0140027285864588 m/s
    Reynolds en tubos: 4208.111323633804

    jh: 0.003337
    jf: 0.0065
    h lado tubos (hi): 395.7529278140253 W/m2K
    ΔP lado tubos: 16925.91091196233 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 0.7485878436101264 m
    Diámetro interno coraza: 0.8137398398506381 m
    Separación de bafles: 0.3254959359402553 m
    Pitch de los tubos: 0.025 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.052973802156811384 m2
    Diámetro hidráulico: 0.014201000000000005 m
    Velocidad lado coraza: 0.6464813539547363 m/s
    Re: 16754.744116207963
    Corte de bafle: 45%

    jh: 0.0035
    jf

In [15]:
U_calculado3 = obj.coefiente_global_U(
    h_conv = [intercambiador3.hi,intercambiador3.h_coraza],
    k = tuberia_acero2.k,
    di = tuberia_acero2.di,
    do = tuberia_acero2.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U,
    valor_calculado = U_calculado3
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador3.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador3.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 220.96905580905903 W/m2K
Para el valor calculado = 220.96905580905903, %error = 24.065616560460814%
Para el valor calculado = 16925.91091196233, %error = 57.685222720094174%
Para el valor calculado = 38671.27084632417, %error = 157.80847230882782%


Esta propuesta no fue buena. Se realiza otra considerando que la caída de presión en los tubos debe aumentar a 40kPa y en la coraza debe disminuir a 15kPa.

## Iteración 4

In [16]:
tuberia_acero3 = obj.Tuberia(
    material = "Acero al carbón",
    do = 0.020,
    espesor = 0.002,
    L = 4,
    k = 43
)

intercambiador4 = obj.Intercambiador()
pasos_tubos = 4
corte_bafle = "15%"

obj.lado_tubos(
    Q = Q,
    U = U_calculado3, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero3, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador4
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero3,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador4.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador4
)

print(intercambiador4)


Para el lado tubos:

LADO TUBOS: 


    Pasos: 4
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 366.5832257292042 m2
    Número de tubos: 1458.5883107343727 tubos
    Longitud tubos: 4 m
    Diámetro exterior: 0.02 m
    Espesor: 0.002 m 
    Tubos por paso: 364.6470776835932 tubos
    velocidad en lado tubos: 1.2993357150206406 m/s
    Reynolds en tubos: 4793.105082076141

    jh: 0.0038
    jf: 0.0058
    h lado tubos (hi): 577.4757863859228 W/m2K
    ΔP lado tubos: 39515.72486748183 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 1.0398187529378193 m
    Diámetro interno coraza: 1.1130145468939618 m
    Separación de bafles: 0.44520581875758475 m
    Pitch de los tubos: 0.025 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.09910411052780572 m2
    Diámetro hidráulico: 0.014201000000000005 m
    Velocidad lado coraza: 0.345561603449911 m/s
    Re: 8955.859603330742
    Corte de bafle: 15%

    jh: 0.0068
    jf: 0.0

In [17]:
U_calculado4 = obj.coefiente_global_U(
    h_conv = [intercambiador4.hi,intercambiador4.h_coraza],
    k = tuberia_acero3.k,
    di = tuberia_acero3.di,
    do = tuberia_acero3.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U_calculado3,
    valor_calculado = U_calculado4
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador4.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador4.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 253.51216284620858 W/m2K
Para el valor calculado = 253.51216284620858, %error = 14.727449921888741%
Para el valor calculado = 39515.72486748183, %error = 1.210687831295418%
Para el valor calculado = 17187.50846532057, %error = 14.583389768803803%


## Iteración 5

In [18]:
intercambiador5 = obj.Intercambiador()
corte_bafle = "20%"

obj.lado_tubos(
    Q = Q,
    U = U_calculado3, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero3, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador5
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero3,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador5.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador5
)

print(intercambiador5)

Para el lado tubos:

LADO TUBOS: 


    Pasos: 4
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 366.5832257292042 m2
    Número de tubos: 1458.5883107343727 tubos
    Longitud tubos: 4 m
    Diámetro exterior: 0.02 m
    Espesor: 0.002 m 
    Tubos por paso: 364.6470776835932 tubos
    velocidad en lado tubos: 1.2993357150206406 m/s
    Reynolds en tubos: 4793.105082076141

    jh: 0.0037
    jf: 0.0058
    h lado tubos (hi): 562.2790551652406 W/m2K
    ΔP lado tubos: 39515.72486748183 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 1.0398187529378193 m
    Diámetro interno coraza: 1.1130145468939618 m
    Separación de bafles: 0.44520581875758475 m
    Pitch de los tubos: 0.025 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.09910411052780572 m2
    Diámetro hidráulico: 0.014201000000000005 m
    Velocidad lado coraza: 0.345561603449911 m/s
    Re: 8955.859603330742
    Corte de bafle: 20%

    jh: 0.00645
    jf: 0.

In [19]:
U_calculado5 = obj.coefiente_global_U(
    h_conv = [intercambiador5.hi,intercambiador5.h_coraza],
    k = tuberia_acero3.k,
    di = tuberia_acero3.di,
    do = tuberia_acero3.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U_calculado3,
    valor_calculado = U_calculado5
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador5.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador5.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 246.78612292156615 W/m2K
Para el valor calculado = 246.78612292156615, %error = 11.683566741044427%
Para el valor calculado = 39515.72486748183, %error = 1.210687831295418%
Para el valor calculado = 14732.150113131918, %error = 1.7856659124538783%


Esta última iteración satisface los requerimentos en caída de presión y de transferencia de calor, entonces este es un diseño que se ajusta al servicio requerido. Sin embargo, es posible observar que el número de tubos (1450) es alto. 

A continuación se realizan otras propuestas con el fin de identificar un diseño, con menos tubos, que satisfaga los requerimentos. Como el número de tubos es función del área total requerida para la transferencia de calor y el área exterior del tubo a emplear $$N_{tubos} = \frac{A_{total}}{A_{ext-tubo}}$$ se evaluarán tuberías más grandes y se hacen los cambios pertinentes para satisfacer las caídas de presión.

## Iteración 6

In [20]:
tuberia_acero4 = obj.Tuberia(
    material = "Acero al carbón",
    do = 0.038,
    espesor = 0.0026,
    L = 7,
    k = 43
)

intercambiador6 = obj.Intercambiador()
pasos_tubos = 4
corte_bafle = "25%"

obj.lado_tubos(
    Q = Q,
    U = U, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero4, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador6
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero4,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador6.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador6
)

print(intercambiador6)


Para el lado tubos:

LADO TUBOS: 


    Pasos: 4
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2
    Número de tubos: 333.1037716322815 tubos
    Longitud tubos: 7 m
    Diámetro exterior: 0.038 m
    Espesor: 0.0026 m 
    Tubos por paso: 83.27594290807038 tubos
    velocidad en lado tubos: 1.3538384140221495 m/s
    Reynolds en tubos: 10238.02693981639

    jh: 0.004
    jf: 0.00475
    h lado tubos (hi): 633.36728933992 W/m2K
    ΔP lado tubos: 32281.01291879852 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 1.0352172937519881 m
    Diámetro interno coraza: 1.108285995405418 m
    Separación de bafles: 0.4433143981621672 m
    Pitch de los tubos: 0.0475 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.09826382780894227 m2
    Diámetro hidráulico: 0.0269819 m
    Velocidad lado coraza: 0.3485166017453802 m/s
    Re: 17161.643176356454
    Corte de bafle: 25%

    jh: 0.0053
    jf: 0.064
    h l

In [21]:
U_calculado6 = obj.coefiente_global_U(
    h_conv = [intercambiador6.hi,intercambiador6.h_coraza],
    k = tuberia_acero4.k,
    di = tuberia_acero4.di,
    do = tuberia_acero4.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U,
    valor_calculado = U_calculado6
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador6.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador6.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 261.62066380553426 W/m2K
Para el valor calculado = 261.62066380553426, %error = 10.095991819404034%
Para el valor calculado = 32281.01291879852, %error = 19.297467703003704%
Para el valor calculado = 14722.28725412223, %error = 1.8514183058518028%


Este diseño no cumple para la caída de presión en tuberías. Se prueba con unas un poco más largas. También se aumenta el espesor de la tubería a 3.2 mm.

### Iteración 7

In [23]:
tuberia_acero5 = obj.Tuberia(
    material = "Acero al carbón",
    do = 0.038,
    espesor = 0.0032,
    L = 7.32,
    k = 43
)

intercambiador7 = obj.Intercambiador()
pasos_tubos = 4
corte_bafle = "25%"

obj.lado_tubos(
    Q = Q,
    U = U, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero5, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador7
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero5,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador7.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador7
)

print(intercambiador7)


Para el lado tubos:

LADO TUBOS: 


    Pasos: 4
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2
    Número de tubos: 318.54185811830195 tubos
    Longitud tubos: 7.32 m
    Diámetro exterior: 0.038 m
    Espesor: 0.0032 m 
    Tubos por paso: 79.63546452957549 tubos
    velocidad en lado tubos: 1.5252934170320172 m/s
    Reynolds en tubos: 11112.609928309928

    jh: 0.004
    jf: 0.0046
    h lado tubos (hi): 713.5792181605117 W/m2K
    ΔP lado tubos: 42577.09490422354 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 1.0151627137880064 m
    Diámetro interno coraza: 1.0876775079428311 m
    Separación de bafles: 0.4350710031771325 m
    Pitch de los tubos: 0.0475 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.09464338890277821 m2
    Diámetro hidráulico: 0.0269819 m
    Velocidad lado coraza: 0.36184857431135864 m/s
    Re: 17818.135736161254
    Corte de bafle: 25%

    jh: 0.0046
    jf: 0.046

In [25]:
U_calculado7 = obj.coefiente_global_U(
    h_conv = [intercambiador7.hi,intercambiador7.h_coraza],
    k = tuberia_acero5.k,
    di = tuberia_acero5.di,
    do = tuberia_acero5.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U,
    valor_calculado = U_calculado7
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador7.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador7.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 260.22065590334427 W/m2K
Para el valor calculado = 260.22065590334427, %error = 10.57709419129063%
Para el valor calculado = 42577.09490422354, %error = 6.442737260558842%
Para el valor calculado = 11928.146822243029, %error = 20.479021185046477%


Este diseño satisface los requerimentos para la transferencia de calor y la caída pero sólo del lado tubos. Como en el lado coraza la caída debe aumentar, se se disminuye el corte de bafle, de 25 a 15%, de forma que se obstruya el flujo por la coraza y aumente la caída de presión.

### Iteración 8

In [27]:
intercambiador8 = obj.Intercambiador()
pasos_tubos = 4
corte_bafle = "15%"

obj.lado_tubos(
    Q = Q,
    U = U, 
    pasos_tubos = pasos_tubos,
    caracteristicas_tuberia = tuberia_acero5, 
    FlujoCaliente = FlujoCaliente, 
    FlujoFrio = FlujoFrio, 
    FlujoTubos = FlujoTubos,
    Intercambiador = intercambiador8
    )

obj.lado_coraza(
    caracteristicas_tuberia = tuberia_acero5,
    FlujoCoraza = FlujoCoraza,
    numero_tubos = intercambiador7.numero_tubos,
    pasos_tubos = pasos_tubos,
    tipo_pitch = tipo_pitch,
    corte_bafle = corte_bafle,
    Intercambiador = intercambiador8
)

print(intercambiador8)

Para el lado tubos:

LADO TUBOS: 


    Pasos: 4
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2
    Número de tubos: 318.54185811830195 tubos
    Longitud tubos: 7.32 m
    Diámetro exterior: 0.038 m
    Espesor: 0.0032 m 
    Tubos por paso: 79.63546452957549 tubos
    velocidad en lado tubos: 1.5252934170320172 m/s
    Reynolds en tubos: 11112.609928309928

    jh: 0.004
    jf: 0.0046
    h lado tubos (hi): 713.5792181605117 W/m2K
    ΔP lado tubos: 42577.09490422354 Pa

LADO CORAZA:

    Diámetro del banco de tubos: 1.0151627137880064 m
    Diámetro interno coraza: 1.0876775079428311 m
    Separación de bafles: 0.4350710031771325 m
    Pitch de los tubos: 0.0475 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.09464338890277821 m2
    Diámetro hidráulico: 0.0269819 m
    Velocidad lado coraza: 0.36184857431135864 m/s
    Re: 17818.135736161254
    Corte de bafle: 15%

    jh: 0.00525
    jf: 0.06

In [28]:
U_calculado8 = obj.coefiente_global_U(
    h_conv = [intercambiador8.hi,intercambiador8.h_coraza],
    k = tuberia_acero5.k,
    di = tuberia_acero5.di,
    do = tuberia_acero5.do,
    foul_f = [propiedadesPetroleo.Rf, propiedadesQueroseno.Rf]
)

error_U = obj.error(
    valor_propuesto=U,
    valor_calculado = U_calculado8
    )

error_presion_tubos = obj.error(
    valor_propuesto = 40e3,
    valor_calculado = intercambiador8.caida_presion_tubos
)

error_presion_coraza = obj.error(
    valor_propuesto = 15e3,
    valor_calculado = intercambiador8.caida_presion_coraza
)

El coeficiente global de transferecia de calor: 271.36712238700466 W/m2K
Para el valor calculado = 271.36712238700466, %error = 6.746693337799087%
Para el valor calculado = 42577.09490422354, %error = 6.442737260558842%
Para el valor calculado = 16336.37499568067, %error = 8.909166637871131%


Este diseño satisface todos los requerimentos del servicio planteado.

## Resultados

#### LADO TUBOS: 


    Pasos: 4
    Número de tubos: 318.54185811830195 tubos
    Tubos por paso: 79.63546452957549 tubos
    Longitud tubos: 7.32 m
    Diámetro exterior: 0.038 m
    Espesor: 0.0032 m 
    Diferencia de temperaturas media: 79.92081901047641
    Área total de transferencia de calor: 278.36271225024535 m2    
    velocidad en lado tubos: 1.5252934170320172 m/s
    Reynolds en tubos: 11112.609928309928

    jh: 0.004
    jf: 0.0046
    h lado tubos (hi): 713.5792181605117 W/m2K
    ΔP lado tubos: 42577.09490422354 Pa

#### LADO CORAZA:

    Diámetro del banco de tubos: 1.0151627137880064 m
    Diámetro interno coraza: 1.0876775079428311 m
    Separación de bafles: 0.4350710031771325 m
    Pitch de los tubos: 0.0475 m
    Tipo de pitch: triangular
    Área de flujo en coraza: 0.09464338890277821 m2
    Diámetro hidráulico: 0.0269819 m
    Velocidad lado coraza: 0.36184857431135864 m/s
    Re: 17818.135736161254
    Corte de bafle: 15%

    jh: 0.00525
    jf: 0.063
    h lado coraza (ho ó hs): 895.1942975377983 W/m2K
    ΔP lado coraza: 16336.37499568067 Pa

    


### Referencias

[1] Sinnott, R. (2005). Chemical Engineering Design: Chemical Engineering Volume 6 (Chemical Engineering Series) (4th ed.). Butterworth-Heinemann.